### PR TITLE
feat: replace mat-input with mat-slider

### PR DESCRIPTION
**Pull request by vitalymatyushik**
Replace the mat-input components for Threshold Min and Threshold Max to <mat-slider> (see [Slider](https://material.angular.io/components/slider/overview)) in the [rule settings dialog component](https://github.com/ghostfolio/ghostfolio/tree/main/apps/client/src/app/components/rule/rule-settings-dialog).

image

### Related Issue
- [[FEATURE] Change input to slider in rule settings dialog](https://github.com/ghostfolio/ghostfolio/issues/3917)

### CODE REVIEW

File: apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html [hunk 1]
```diff
@@ -1,32 +1,37 @@
 <div mat-dialog-title>{{ data.rule.name }}</div>
 
 <div class="py-3" mat-dialog-content>
-  <mat-form-field
+  <div
     appearance="outline"
```
Comment (dtslvr) [inline apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html:5]: Remove please

File: apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html [hunk 2]
```diff
@@ -1,32 +1,37 @@
...
+      [max]="data.rule.configuration.threshold.max"
+      [min]="data.rule.configuration.threshold.min"
+      [step]="data.rule.configuration.threshold.step"
+    >
+      <input matSliderThumb [(ngModel)]="data.settings.thresholdMin" />
+    </mat-slider>
+    <mat-label>{{ data.settings.thresholdMin }}</mat-label>
+  </div>
+  <div
     appearance="outline"
```
Comment (dtslvr) [inline apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html:21]: Remove please

File: apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html [hunk 3]
```diff
@@ -1,32 +1,37 @@
 <div mat-dialog-title>{{ data.rule.name }}</div>
 
 <div class="py-3" mat-dialog-content>
-  <mat-form-field
+  <div
     appearance="outline"
     class="w-100"
     [ngClass]="{ 'd-none': !data.rule.configuration.thresholdMin }"
   >
-    <mat-label i18n>Threshold Min</mat-label>
```
Comment (dtslvr) [inline apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html:9]: `<h5 i18n>Threshold Min</h5>`

File: apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html [hunk 4]
```diff
@@ -1,32 +1,37 @@
...
+      <input matSliderThumb [(ngModel)]="data.settings.thresholdMin" />
+    </mat-slider>
+    <mat-label>{{ data.settings.thresholdMin }}</mat-label>
+  </div>
+  <div
     appearance="outline"
     class="w-100"
     [ngClass]="{ 'd-none': !data.rule.configuration.thresholdMax }"
   >
     <mat-label i18n>Threshold Max</mat-label>
```
Comment (dtslvr) [inline apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html:25]:
```
<h5 i18n>Threshold Max</h5>
<label>{{ data.rule.configuration.threshold.min }}</label>
```

File: apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html [hunk 5]
```diff
@@ -1,32 +1,37 @@
...
-      type="number"
-      [(ngModel)]="data.settings.thresholdMax"
-    />
-  </mat-form-field>
+      [max]="data.rule.configuration.threshold.max"
+      [min]="data.rule.configuration.threshold.min"
+      [step]="data.rule.configuration.threshold.step"
+      ><input matSliderThumb [(ngModel)]="data.settings.thresholdMax"
+    /></mat-slider>
+    <mat-label>{{ data.settings.thresholdMax }}</mat-label>
```
Comment (dtslvr) [inline apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html:33]: `<label>{{ data.rule.configuration.threshold.max }}</label>`

File: apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html [hunk 6]
```diff
@@ -1,32 +1,37 @@
...
-  <mat-form-field
+  <div
     appearance="outline"
     class="w-100"
     [ngClass]="{ 'd-none': !data.rule.configuration.thresholdMin }"
   >
-    <mat-label i18n>Threshold Min</mat-label>
-    <input
-      matInput
+    <label i18n>Threshold Min</label>
```
Comment (dtslvr) [inline apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html:9]:
```
<label>{{ data.rule.configuration.threshold.min }}</label>
```

File: apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html [hunk 7]
```diff
@@ -1,32 +1,37 @@
...
-    />
-  </mat-form-field>
-  <mat-form-field
+      [max]="data.rule.configuration.threshold.max"
+      [min]="data.rule.configuration.threshold.min"
+      [step]="data.rule.configuration.threshold.step"
+    >
+      <input matSliderThumb [(ngModel)]="data.settings.thresholdMin" />
+    </mat-slider>
+    <mat-label>{{ data.settings.thresholdMin }}</mat-label>
```
Comment (dtslvr) [inline apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html:18]:
```
<label>{{ data.rule.configuration.threshold.max }}</label>
```

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thank you for the contribution, @vitalymatyushik! 😎
I have reviewed the code and added some comments. The logic is already fine, I would introduce titles and use the labels to indicate min and max.

File: apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html (hunk 7 above)
Comment (vitalymatyushik) [reply apps/client/src/app/components/rule/rule-settings-dialog/rule-settings-dialog.html]: please correct me if I'm wrong: just to add the label but don't change the value to `data.rule.configuration.threshold.max`

**Comment by vitalymatyushik** [comment]
thank you @dtslvr for review - I've updated the PR (wanted only clarify if I understood comment correctly)
here is the result of the change 
image

**Comment by dtslvr** [comment]
> thank you @dtslvr for review - I've updated the PR

Thank you, @vitalymatyushik.

> here is the result of the change

I have created a mockup to illustrate what I have in mind:

376326597-ee95bd98-91bd-4e54-ae4a-ab7a58ef40e0

**Comment by vitalymatyushik** [comment]
updated 
image

**Review by dtslvr — APPROVED** [review]
Thank you for this contribution, @vitalymatyushik! 🙌🏻 
I have made some minor adjustments. Ready to merge now.

**Pull request merged**